### ISSUE TITLE
SHOULD throw if ce-datacontenttype is present in binary mode?

### ISSUE THREAD

**Issue by c-pius** [cloudevents/sdk-javascript#381]
**Describe the Bug**
[Spec](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#311-http-content-type) states that the `ce-datacontenttype` header MUST NOT be present in a binary mode message. If it is passed anyway, no error is thrown and the given `ce-datacontenttype` is treated as extension attribute overwriting the value from the `Content-Type` header.

**Steps to Reproduce**
Use the following snippet:

```js
const event = HTTP.toEvent({
  headers: {
    'Content-Type': 'application/json',
    'ce-id': '0815',
    'ce-specversion': '1.0',
    'ce-type': 'my.event.type',
    'ce-source': 'my.event.source',
    'ce-datacontenttype': 'text/xml',
  },
  body: {}
})
```

Check the resulting event:

```js
{
  //...
  id: '0815',
  specversion: '1.0',
  type: 'my.event.type',
  source: 'my.event.source',
  datacontenttype: 'text/xml'
}
```

**Expected Behavior**
`HTTP.toEvent` should throw to prevent this.

**Additional context**
Not quite sure why the `datacontenttype` (also without `ce-` prefix) is part of the binary parsers so maybe it is intentional though?
https://github.com/cloudevents/sdk-javascript/blob/6be3b2751401e529fce6ccf8a406a057472c10a9/src/message/http/headers.ts#L98

And I assume this is also a bug (where it should use `v03binaryParsers` on the else path instead):
https://github.com/cloudevents/sdk-javascript/blob/6be3b2751401e529fce6ccf8a406a057472c10a9/src/message/http/index.ts#L141

**Comment by lance** [comment]
>  `HTTP.toEvent` should throw to prevent this.

I don't think it should, to be honest. This is similar to other scenarios that have been pointed out with regard to receiving malformed events. The `HTTP.toEvent()` function is meant to be used when receiving events via incoming HTTP requests, and we should be as forgiving as possible. To me the debate is really, what should the `datacontenttype` value be in this scenario, `text/xml` or `application/json`?

Thanks for pointing out those two bugs. You're right that they are incorrect.
